hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azure/StorageInterfaceImpl.java

```diff
@@ -435,6 +456,24 @@ public void startCopyFromBlob(CloudBlobWrapper sourceBlob, BlobRequestOptions op
           null, dstAccessCondition, options, opContext);
     }
 
+    @Override
+    public void startCopyFromBlob(CloudBlobWrapper sourceBlob, BlobRequestOptions options,
+                                  OperationContext opContext, boolean overwriteDestination, String eTag)
+            throws StorageException, URISyntaxException {
+      AccessCondition dstAccessCondition =
+              overwriteDestination
+                      ? null
+                      : AccessCondition.generateIfNotExistsCondition();
+      if (dstAccessCondition != null) {
```

Comment:
Better we check if etag == null. Agree sdk checks that before populating header. But SDK can get change in future.

Reply:
Checked in the calling flow already.

Reply:
Checked in which flow? etag is a field in the method, and can be null. I am suggesting we do not call SDK's setifMatch() if etag with us is null.

Reply:
Done